web/src/test/java/no/nav/modiapersonoversikt/testutils/TestEnvironmentExtension.kt

```diff
@@ -0,0 +1,29 @@
+package no.nav.modiapersonoversikt.testutils
+
+import org.junit.jupiter.api.extension.AfterTestExecutionCallback
+import org.junit.jupiter.api.extension.BeforeTestExecutionCallback
+import org.junit.jupiter.api.extension.ExtensionContext
+
+class TestEnvironmentExtension(val testEnvironment: Map<String, String?>) :
+    BeforeTestExecutionCallback, AfterTestExecutionCallback {
+
+    private val originalEnvironment: Map<String, String?> = testEnvironment.keys.associateWith(System::getProperty)
+
+    override fun beforeTestExecution(context: ExtensionContext) {
+        testEnvironment.setAsEnvironment()
+    }
+
+    override fun afterTestExecution(context: ExtensionContext) {
+        originalEnvironment.setAsEnvironment()
+    }
```

Comment:
Blir disse kjørt opp automatisk når man kjører andre tester? De trenger ikke å kalles?

Reply:
`TestEnvironmentExtension` er en utvidelse til junit for å sette properties slik at `EnvironmentUtils.getRequiredProperty` fungerer.

Funksjonene må ikke kalles manuelt, men fungerer med som ett slags "middleware" for hver funksjon annotert med `@Test`. 
For å sette det opp trenger man bare å legge ved følgende i test-filen hvor man har behov for å sette properties:
```kotlin
    @JvmField
    @RegisterExtension
    val testenvironment = TestEnvironmentExtension(
        mapOf(
            "NAIS_APP_NAME" to "appname",
            "NAIS_CLUSTER_NAME" to "cluster",
            "NAIS_NAMESPACE" to "namespace"
        )
    )
```

JUnit passer dermed på at funksjonene `beforeTestExecution` og `afterTestExecution` blir kjørt før og etter hver enkelt test.